### PR TITLE
Added GitHub Actions

### LINKED ISSUE #2: Github Actions
PySyft has a list of GitHub action that makes sure the PRs respect a given "code quality" standard + tested against different versions of python + torch.

We should be on the same level as Syft regarding GitHub Actions

### PR DESCRIPTION
**Pull request by aanurraj**
## Description
fixes #2 
## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 --config=.flake8 .
+          black --check .
+          # Run type checker in CI such that type errors can be identified and gradually addressed.
+          # Once all existing type errors are addressed, the `|| echo` after the mypy call can be removed.
+          mypy syft || echo "Type errors found, continuing build..."
+      - name: Check docstring coverage
+        run: |
+          interrogate -mvp syft/ -f 52.5
```
Comment (gmuraru) [inline .github/workflows/tests.yml:67]: I think here should be something else :3

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Install build tools
+        run: |
+          pip3 install flake8
+          #- pip3 install flake8-docstrings
+          pip3 install flake8-comprehensions
+          pip3 install pep8-naming
+      - name: Lint, format, and type-check
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 --config=.flake8 .
```
Comment (gmuraru) [inline .github/workflows/tests.yml:60]: Q: Shouldn't we also need a flake8 file?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          # Run type checker in CI such that type errors can be identified and gradually addressed.
+          # Once all existing type errors are addressed, the `|| echo` after the mypy call can be removed.
+          mypy sympc || echo "Type errors found, continuing build..."
+      - name: Check docstring coverage
+        run: |
+          interrogate -mvp syft/ -f 52.5
+      - name: Scan for security issues
+        run: |
+          pip install bandit
+          bandit -r syft -ll
```
Comment (arturomf94) [inline .github/workflows/tests.yml:71]: I thini this `sydt` should probably change too.

**Comment by arturomf94** [comment]
We're not actually running any tests on this workflow. Will we include that? :)

**Comment by aanurraj** [comment]
> We're not actually running any tests on this workflow. Will we include that? :)

Yes currently we don't have any tests, as soon as tests are merged Ill be adding that too :)

File: .github/workflows/tests.yml (hunk 3 above)
Comment (aanurraj) [reply .github/workflows/tests.yml:71]: oops forgot to push the changes

Comment (aanurraj) [reply .github/workflows/tests.yml]: done

**Review by arturomf94 — APPROVED** [review]

File: .flake8 [hunk 4]
```diff
@@ -0,0 +1,15 @@
+[flake8]
+ignore = N801, N802, N805, N806, N812, N803, N804
```
Comment (gmuraru) [inline .flake8:2]: Q: Can't we remove some of those?

**Pull request merged**